src/GafferImage/ImageWriter.cpp

```diff
@@ -841,6 +874,43 @@ const std::string ImageWriter::currentFileFormat() const
 	}
 }
 
+void ImageWriter::registerDefaultColorSpace( DefaultColorSpaceFunction f )
+{
+	defaultColorSpaceFunc() = f;
+}
+
+ImageWriter::DefaultColorSpaceFunction &ImageWriter::defaultColorSpaceFunc()
+{
+	static DefaultColorSpaceFunction g_colorSpaceFunction;
+	return g_colorSpaceFunction;
+}
+
+const std::string ImageWriter::defaultColorSpace() const
+{
+	const std::string fileFormatName = currentFileFormat();
+	const ValuePlug *optionsPlug = this->getChild<ValuePlug>( fileFormatName );
+
+	if( optionsPlug == NULL)
+	{
+		return "";
```

Comment:
could we? it seems to me that without data type, you cant make a good assumption for the colorspace.
Take for example a tiff file, if you are writing an 8 bits file, using linear is a terrible idea.